Azure/communication-ui-library-android
# [Bug] Fix AudioSessionManager life cycle

## Purpose

* Bluetooth SCO doesn't disable after a call

This PR cleans the life cycle of audio session manager to better align with the activities life-cycle.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
- Connect Bluetooth Headset (with high quality stereo audio support, Good: galaxy buds, sony wireless, airpods, etc. BAD: like Jabra or mono-headset) (prior to call)
- Play some music (observe good quality), leave running during call
- Setup screen should switch to low-quality (Bluetooth SCO, with mic support/bidirectional protocol, not A2DP audio stream).
- Leaving Setup/Call should revert to high-quality, if music sounds bad upon leaving. You should audibly notice the change between A2DP/SCO when joining/leaving a call.

## What to Check
Verify that the following are valid
* High quality audio persists after call is ended.

## Other Information
- Requires A2DP/AptX support on headset
